tests/e2e/infodocs.js

```diff
@@ -0,0 +1,322 @@
+      }).then(newInfoDocs => {
+        // ...which means that the first two latest_replication_date values should change...
+        assert.notEqual(newInfoDocs[0].latest_replication_date, infoDocs[0].latest_replication_date);
+        assert.notEqual(newInfoDocs[1].latest_replication_date, infoDocs[1].latest_replication_date);
+        // ..and the last should not.
+        assert.equal(newInfoDocs[2].latest_replication_date, infoDocs[2].latest_replication_date);
+
+        docs.pop(); // (remove the failing one, we're done with it)
+
+        // When we delete a doc...
+        docs[1]._deleted = true;
+
+        return utils.requestOnTestDb({
+          path: '/_bulk_docs',
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: {docs: docs}
+        });
+      }).then(result => {
+        // ...everything should work...
+        assert.isTrue(result[0].ok);
+        assert.isTrue(result[1].ok);
+
+        // ...and the second doc should not have a infodoc anymore
+        return delayedInfoDocsOf(docs[1]._id);
+      }).then(results => {
+        assert.isNull(results[0]);
+      });
+    });
+  });
+
+  describe('legacy data support', () => {
+    it('finds and migrates data from the medic doc', () => {
+      const testDoc = {
+        _id: 'yuiop',
+        data: 'data',
+        transitions: {
+          some: 'transition info'
+        }
+      };
+
+      // Prepare an existing document
+      return utils.stopSentinel()
+        .then(() => utils.db.post(testDoc))
+        .then(({id, rev}) => {
+          testDoc._id = id;
+          testDoc._rev = rev;
+
+          // Skip the preceeding write in sentinel
+          return utils.setProcessedSeqToNow();
+        })
+        .then(utils.startSentinel)
+        .then(() => {
+          // Now update the document via api and with sentinel ready to roll
+
+          testDoc.data = 'data changed';
+
+          // Should be through api
+          return utils.saveDoc(testDoc);
+        })
+        .then(() => delayedInfoDocsOf(testDoc._id))
+        .then(([infodoc]) => {
+          assert.isOk(infodoc.initial_replication_date, 'expected an initial_replication_date');
+          assert.isOk(infodoc.latest_replication_date, 'expected a latest_replication_date');
+          assert.deepEqual(infodoc.transitions, {
+            some: 'transition info'
+          });
+        });
+    });
+    it('finds and migrates data from the medic infodoc', () => {
+      // In legacy situations the transition info was on the doc, while other
+      // information was on the infodoc
+      const testDoc = {
+        data: 'data',
+        transitions: {
+          some: 'transition info'
+        }
+      };
+      const legacyInfodoc = {
+        type: 'info',
+        some: 'legacy data'
+      };
+
+      // Prepare an existing document
+      return utils.stopSentinel()
+        .then(() => utils.db.post(testDoc))
```

Comment:
By using `db.post` you're bypassing the API controller that would create the "initial" sentinel infodoc. I think it would be essential for the test to not bypass API.

Reply:
Ugh, I completely missed the line where you update via API! Please disregard!